react/src/components/home/Home.jsx

```diff
@@ -25,55 +25,185 @@ function Home() {
+    e.preventDefault();
+
+    setQuery("");
+    clearQueryField();
+
+    if (locationFilter) {
+      resultsData.firstFilteredResults = filterByLocation(
+        results,
+        locationFilter
+      );
+    } else if (!locationFilter) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onResetLocationFilter = (e) => {
+    e.preventDefault();
+
+    setLocationFilter("");
+    clearLocationField();
+
+    if (query) {
+      resultsData.firstFilteredResults = filterByValue(results, query);
+    } else if (!query) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  function filterByValue(array, string) {
+    console.log(string);
+
+    return array.filter((o) =>
+      Object.keys(o).some((k) =>
+        String(o[k]).toLowerCase().includes(string.toLowerCase())
+      )
+    );
+  }
+
+  function filterByLocation(array, string) {
+    return array.filter(
+      (result) => result.city == string || result.zipcode == string
+    );
+  }
+
+  function clearQueryField() {
+    document.getElementById("searchFilter").value = "";
+  }
+
+  function clearLocationField() {
+    document.getElementById("searchLocation").value = "";
+  }
+
   useEffect(() => {
     navigator.geolocation.getCurrentPosition((position) => {
       if (position.coords.latitude) {
         setZoom(16);
         setCurrent((prevState) => {
           const newCurrent = { ...prevState };
-          
+
           newCurrent.lat = position.coords.latitude;
           newCurrent.lng = position.coords.longitude;
           newCurrent.active = "on";
 
           return { ...newCurrent };
         });
-        
+
         setCenter((prevState) => {
           const newCurrent = { ...prevState };
-          
+
           newCurrent.lat = position.coords.latitude;
           newCurrent.lng = position.coords.longitude;
-          
+
           return { ...newCurrent };
         });
       }
     });
   }, []);
 
-  const SearchBar = () => {
-    return (
-      <>
-        <input
-          id="searchFilter"
-          type="search"
-          className={styles.searchInputLeft}
-          placeholder=" Search what you need"
-        />
-        <input
-          id="searchLocation"
-          type="search"
-          className={styles.searchInputRight}
-          placeholder=" Please Enter a City or Zip Code"
-        />
-        <i className={`fa-solid fa-magnifying-glass ${styles.searchIcon}`} onClick={() => {}} />
-      </>
-    );
-  };
-
```

Comment:
Why did you remove and move this? This was initially pulled out so when the logic was added it could be moved to its own component to declutter Home.jsx.

Reply:
I tried using it with the const SearchBar but it was preventing the passing of information to the card for rendering the results somehow. Removing it and just adding the code within without the function worked better for the card rendering.

Reply:
It mau be how you were passing the information. I still think you should create your own SearchBar component and move all the logic to that component. Like how the SearchResults, and FilterButtons components were created and the logic to do with them were moved with them.